[SPEC-8103] Update GameLift server instructions for new exposed API (#869)
o3de/o3de.org: content/docs/user-guide/gems/reference/aws/aws-gamelift/cpp-api.md

To search for sessions, call `AWSGameLiftClientManager::SearchSessions()` or `AWSGameLiftClientManager::SearchSessionsAsync()` and pass in a reference to the search request, which contains the search criteria. When the search is completed, you can iterate through `SessionConfigs` from `SearchSessionsResponse`.

```cpp
// For example, make synchronous call to search active game sessions on specific fleet
AWSGameLift::AWSGameLiftSearchSessionsRequest request;
request.m_fleetId = "YourGameLiftFleetId";
AzFramework::SearchSessionsResponse result;
AWSGameLift::AWSGameLiftSessionRequestBus::BroadcastResult(result, &AWSGameLift::AWSGameLiftSessionRequestBus::Events::SearchSessions, request);

> // For example, make asynchronous call to search active game sessions on specific fleet and get response from notification

Reviewer: nit: "on a specific fleet" ?

To begin the process that allows a player to join the game, call `AWSGameLiftClientManager::JoinSession()` or `AWSGameLiftClientManager::JoinSessionAsync()`, and pass in the game session id and the player id that will join. The process returns `true` if both steps, reserving player slot and initializing connection, succeed. If either step fails, the process returns `false`.

```cpp

> // For example, make synchronous call to join specific session

Reviewer: Nit: "join a specific session" ？

AWSGameLift::AWSGameLiftJoinSessionRequest request;
request.m_sessionId = "YourGameSessionId";
request.m_playerId= "YourPlayerId";
bool result = false;
AWSGameLift::AWSGameLiftSessionRequestBus::BroadcastResult(result, &AWSGameLift::AWSGameLiftSessionRequestBus::Events::JoinSession, request);

> // For example, make asynchronous call to join specific session and get response from notification

Reviewer: Nit: "join a specific session" ？

To leave the game session, call `AWSGameLiftClientManager::LeaveSession()` or `AWSGameLiftClientManager::LeaveSessionAsync()`.

```cpp

> // For example, make synchronous call to leave current session

Reviewer 2: nit: "leave the current session"

AWSGameLift::AWSGameLiftSessionRequestBus::Broadcast(&AWSGameLift::AWSGameLiftSessionRequestBus::Events::LeaveSession);

> // For example, make asynchronous call to leave current session and get notification after process is complete

Reviewer 2: nit: leave the current session and get a notification after the process is complete. 

I would also be wary of "process" here as GameLift has definition of process. Maybe "and get a notification, once the leaving the session is completed"?

## Server notifications

You must notify Amazon GameLift service that your server process is ready to host game sessions, handle requests, and take connections.

To send a notification that your server process is ready, use `AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady()`.

> Put this call in a proper place on your server side, like server system component activate step, and call it after you connect your project to `AzFramework::SessionNotificationBus`.

Reviewer 2: nit: "proper place" seems to indicate there is a definite place. Maybe "Call process ready once your have completed any relevant initialization. One recommended sport is in the [ServerSystemCompontent]'s activate step, after connecting to the `AzFramework::SessionNotificationBus`

> We recommended placing the call after connecting to `AzFramework::SessionNotificationBus` in the `YourProjectServerSystemComponent` activate step.

Reviewer 3: ```suggestion
We recommend placing the call after connecting to `AzFramework::SessionNotificationBus` in the `YourProjectServerSystemComponent` activate step.
```

> To send a notification that your server process is ready, use `AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady()`.
> Call process ready once your have completed any relevant initialization. One recommended place is in `YourProjectServerSystemComponent` activate step, after connecting to `AzFramework::SessionNotificationBus`.

Reviewer 3: Update to improve flow and use active voice:
```suggestion
To send a notification that your server process is ready, complete any relevant initialization and then use `AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady()`. We recommend placing the call after connecting to `AzFramework::SessionNotificationBus`, in the `YourProjectServerSystemComponent` activation step.

Example call:
```

```cpp
AWSGameLift::AWSGameLiftServerRequestBus::Broadcast(&AWSGameLift::AWSGameLiftServerRequestBus::Events::NotifyGameLiftProcessReady);
```

> Once your session is created, it broadcasts the following notifications through `AzFramework::SessionNotificationBus` on server side. You can program how your session responds to these notifications.  

Reviewer 2: nit: "Once the game session is created, the a notification is broadcast through... You can then listen to the notification to {insert example of what customer should do here}"

> Once the game session is created, following notifications are broadcast through `AzFramework::SessionNotificationBus`. You can program how your session responds to these notifications.

Reviewer 3: ```suggestion
After the game session has been created, notifications are broadcast through `AzFramework::SessionNotificationBus`. You can program how your session responds to these notifications.
```
